webapp/public/javascripts/controllers.js

```diff
@@ -60,4 +60,67 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
                     $scope.errorMessages = data;
                 })
         }
+    }]);
+
+lunchrControllers.controller('MapController', ['$scope',
+    function ($scope) {
+
+        $scope.map = { center: { latitude: 49.8651559, longitude: -97.11077669999997 }, zoom: 14 };
+        $scope.getUserLocation = function ( onSuccess, onError )
+        {
+            if ( navigator.geolocation )
+            {
+                navigator.geolocation.getCurrentPosition( function ( position )
+                {
+                    $scope.position = position;
+                    $scope.map = { center: { latitude: position.coords.latitude, longitude: position.coords.longitude }, zoom: 14 };
+                    $scope.marker = {id: 0, coords:{ latitude: position.coords.latitude, longitude: position.coords.longitude }};
+
+                    $scope.$apply( function( )
+                    {
+                        $scope.position = position;
+                        onSuccess( position );
+                    } );
+                },
+                function( error )
+                {
+                    onError( error );
+                } );
+            }
+            else
+            {
+                alert( "Your browser does not support geolocation." );
+            }
+        }
+        
+        function onSuccess( position )
+        {
+            $scope.map = { center: { latitude: position.coords.latitude, longitude: position.coords.longitude }, zoom: 14 };
+            $scope.marker = {id: 0, coords: { latitude: position.coords.latitude, longitude: position.coords.longitude }};
+        }
+        
+        function onError( error )
+        {
+            var errorMessage = "";
+            
+            switch( error.code )
```

Comment:
Error is the value returned from geolocation when an error occurs. There is an optional parameter that is used to hold a function in case an error occurs. I pass that to the onError function. Basically the value is just an error code. Sample code and more info here: http://www.w3schools.com/html/html5_geolocation.asp